### DISCUSSION TITLE
Trying to assign workspace to output; getting "ERROR: Expected one of these tokens..."

### DISCUSSION THREAD

**Discussion by rlue** [i3/i3#5499] [Q&A] [answered]
I'm following the syntax outlined in the docs to [assign a particular workspace to a given output](https://i3wm.org/docs/userguide.html#workspace_screen):

#### Docs

> Examples:
>
> ```
> workspace 1 output LVDS1
> workspace 2 output primary
> workspace 5 output VGA1 LVDS1
> workspace "2: vim" output VGA1
> ```

#### Me

```sh
rlue@rlue-lycra:~/.config $ i3-msg 'workspace "10:零" output DP3'
ERROR: Your command: workspace "10:零" output DP3
ERROR:                                  ^^^^^^^^^^
ERROR: Expected one of these tokens: <end>, '[', 'move', 'exec', 'exit', 'restart', 'reload', 'shmlog', 'debuglog', 'border', 'layout', 'append_layout', 'workspace', 'focus', 'kill', 'open', 'fullscreen', 'sticky', 'split', 'floating', 'mark', 'unmark', 'resize', 'rename', 'nop', 'scratchpad', 'swap', 'title_format', 'title_window_icon', 'mode', 'bar', 'gaps'
[{"success":true},{"success":false,"parse_error":true,"error":"Expected one of these tokens: <end>, '[', 'move', 'exec', 'exit', 'restart', 'reload', 'shmlog', 'debuglog', 'border', 'layout', 'append_layout', 'workspace', 'focus', 'kill', 'open', 'fullscreen', 'sticky', 'split', 'floating', 'mark', 'unmark', 'resize', 'rename', 'nop', 'scratchpad', 'swap', 'title_format', 'title_window_icon', 'mode', 'bar', 'gaps'","input":"workspace \"10:零\" output DP3","errorposition":"                   ^^^^^^^^^^"}]
```

I have also tried putting these lines directly in my config file, and tried variants of the name for the target output (`2`, `secondary`), all with the same result. What am I missing?

(Many thanks in advance.)

**Comment by okraits** [comment, marked as answer]
_i3-msg_ is for commands, not for config directives. You can only put those assignments in your config file - but without _i3-msg_.

**Reply by rlue** [reply]
Thanks, apparently I did not understand the intended behavior so it looked like the config line was not doing anything when in fact it was.

**Reply by rlue** [reply]
To clarify for anyone who might stumble upon this issue in the future:

#### What I thought the intended behavior was

On startup, make the specified workspace active/visible (or at least show a workspace button for the specified workspace in i3bar) on the specified display.

#### What the intended behavior actually is

Assuming there are no windows in the specified workspace already, it will not be active or appear in i3bar. But whenever it is used/invoked (either by switching to it, or by moving a window to it), always assign it to the specified display.
